doc/api/permissions.md

```diff
@@ -0,0 +1,457 @@
+# Permissions
+
+Permissions can be used to control what system resources the
+Node.js process has access to or what actions the process can take
+with those resources. Permissions can also control what modules can
+be accessed by other modules.
+
+* [Process-based permissions](#process-based-permissions) control the Node.js
+  process's access to resources such as the file system or the network.
+  The resource can be entirely allowed or denied, or actions related to it can
+  be controlled; for example, you can allow file system reads while denying
+  writes.
+
+* [Module-based permissions](#module-based-permissions) control which files
+  or URLs are available to other modules during application execution.
+  This can be used to control what modules can be accessed by third-party
+  dependencies, for example.
+
+Both types of permissions can be used together to provide a
+safer environment. If you find a potential security vulnerability,
+please refer to our [Security Policy][].
+
+## Process-based permissions
+
+// STUB
+
+## Module-based permissions
+
+## Policies
+
+<!--introduced_in=v11.8.0-->
+
+<!-- type=misc -->
+
+> Stability: 1 - Experimental
+
+<!-- name=policy -->
+
+Node.js contains experimental support for creating policies on loading code.
+
+Policies are a security feature intended to allow guarantees
+about what code Node.js is able to load. The use of policies assumes
+safe practices for the policy files such as ensuring that policy
+files cannot be overwritten by the Node.js application by using
+file permissions.
```

Comment:
```suggestion
Node.js contains experimental support for controlling what code
modules can load.

Module-based permissions are a security feature intended to allow
guarantees about what code Node.js is able to load. The use of
module-based permissions assumes safe practices for the
policy files such as ensuring that policy files cannot be
overwritten by the Node.js application by using file permissions.
```

We should continue the renaming throughout the text, though perhaps not for the API itself (so like `policy.json` and `--policy` would stay, unless you want to update the code too to rename them everywhere).

Reply:
Not quite sure about this change. That would rename the `policy` mechanism to `Module-based permissions` which is not the intention here. I believe the `Module-based permissions` should be only the namespace on which the `policy` lives.

Reply:
Okay, we can punt on this for now.